### ISSUE TITLE
Apparent downloader corruption with specific large files

### ISSUE THREAD

**Issue by Al12rs** [Nexus-Mods/NexusMods.App#439]
Using the app to download this file:
https://www.nexusmods.com/cyberpunk2077/mods/2155?tab=files&file_id=46864&nmm=1

7zip later extracts with following errors:
```
Extracting archive: C:\Users\Al12r\AppData\Local\Temp\NexusMods.App-c24ec45d-aecf-414d-952c-fee74fe5e6ef\0b10d5d8-7a16-415d-bba1-56684a65ccfe.tmp

ERRORS:
Headers Error


WARNINGS:
There are data after the end of archive

--
Path = C:\Users\Al12r\AppData\Local\Temp\NexusMods.App-c24ec45d-aecf-414d-952c-fee74fe5e6ef\0b10d5d8-7a16-415d-bba1-56684a65ccfe.tmp
Type = Rar5
ERRORS:
Headers Error
WARNINGS:
There are data after the end of archive
Physical Size = 1110369436
Tail Size = 8
Solid = -
Blocks = 4
Encrypted = -
Multivolume = -
Volumes = 1



Archives with Errors: 1

Warnings: 1

Open Errors: 1
```

Install fails with either IOException or Could not find analysis data for archive

The file in question is quite large (1gb+), so there is more time for things to go wrong during download.

**Comment by halgari** [comment]
I took a look at the downloader code and it's pretty inpenetrable. I think we're going to need to rework it into something a bit simpler and implement more tests around this in the process.

**Comment by erri120** [comment]
Powerful downloader library: https://github.com/bezzad/Downloader

**Comment by erri120** [comment]
Was this fixed in #481?

**Comment by Sewer56** [comment]
Kinda, it doesn't corrupt anymore but we've run into a new bug yesterday which may or may not be patched on an unmerged branch at the time I write this (I just woke up).

**Comment by halgari** [comment]
Corruption is fixed as of #481 but we still have some failing tests, not related to the downloader. Closing this and moving work to #486
